### PR TITLE
[Merged by Bors] - chore: bump dependencies (upstream LazyList)

### PR DESCRIPTION
**Pull request by kim-em**
Also adjusts names arguments to `LawfulMonad.mk'` to match their actual order.

For some reason the motive now has to be specified explicitly.
I'm not really interested in understanding why, however, because this file, in Batteries and Mathlib is about to be deprecated on the grounds that LazyList is not currently being used, is not actually useful in practice without VM support (it was ported from Mathlib3, where it did have such support), and MLList supplies much of the functionality.

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/LazyList/Basic.lean [hunk 1]
```diff
@@ -86,98 +69,32 @@ instance : LawfulTraversable LazyList := by
...
-    | case2 _ _ ih =>
-      simp only [LazyList.bind, LazyList.traverse, Seq.seq, Id.map_eq, append, Thunk.pure]
-      rw [← ih]
-      simp [Thunk.pure, Thunk.get, append])
+  (id_map := by
+    intro α xs
+    induction xs using LazyList.rec (motive_2 := fun xs => id <$> xs.get = xs) with
+    | nil => simp only [Functor.map, comp_id, LazyList.bind]
+    | cons h t _ => simp only [Functor.map, comp_id, bind_singleton]
+    | mk f _ => ext; simp_all)
```
Comment (eric-wieser) [inline Mathlib/Data/LazyList/Basic.lean:97]: Any particular reason you swapped the order of these?

Comment (semorrison) [reply Mathlib/Data/LazyList/Basic.lean:97]: Matches the order of the arguments.

**Comment by eric-wieser** [comment]
bors d+

I edited the PR description to capture your zulip comment.

**Comment by eric-wieser** [comment]
Oh, I see CI is done

bors merge

**Pull request closed without merging**